codecarbon/core/emissions.py

```diff
@@ -78,78 +78,98 @@ def get_private_infra_emissions(self, energy: Energy, geo: GeoMetadata) -> float
         :param geo: Country and region metadata
         :return: CO2 emissions in kg
         """
-        compute_with_energy_mix: bool = geo.country_iso_code.upper() != "USA" or (
-            geo.country_iso_code.upper() == "USA" and geo.region is None
+        compute_with_regional_data: bool = (geo.region is not None) and (
+            geo.country_iso_code.upper() in ["USA", "CAN"]
         )
 
-        if compute_with_energy_mix:
-            return self.get_country_emissions(energy, geo)
-        else:
+        if compute_with_regional_data:
             return self.get_region_emissions(energy, geo)
+        else:
+            return self.get_country_emissions(energy, geo)
 
     def get_region_emissions(self, energy: Energy, geo: GeoMetadata) -> float:
         """
-        Computes emissions for a country on private infra,
-        given emissions per unit power consumed at a regional level
+        Computes emissions for a region on private infra.
+        Given an quantity of power consumed, use regional data
+         on emissions per unit power consumed or the mix of energy sources.
         https://github.com/responsibleproblemsolving/energy-usage#calculating-co2-emissions
         :param energy: Mean power consumption of the process (kWh)
         :param geo: Country and region metadata.
         :return: CO2 emissions in kg
         """
+        try:
+            country_emissions_data = self._data_source.get_country_emissions_data(
+                geo.country_iso_code.lower()
+            )
 
-        country_emissions_data = self._data_source.get_country_emissions_data(
-            geo.country_iso_code.lower()
-        )
+            if geo.region not in country_emissions_data:
+                # TODO: Deal with missing data, default to something
+                raise Exception(
+                    f"Region: {geo.region} not found for Country with ISO CODE : {geo.country_iso_code}"
+                )
```

Comment:
This exception is not catched by the following `except DataSourceException`, is that expected?